shell/common/thread_host.h

```diff
@@ -22,6 +27,43 @@ struct ThreadHost {
     Profiler = 1 << 4,
   };
 
+  /// The collection of all the thread configures, and we create custom thread
+  /// configure in engine to info the thread.
+  struct ThreadHostConfig {
+    ThreadHostConfig() : type_mask(0) {}
+
+    explicit ThreadHostConfig(
+        const std::string& name_prefix,
+        uint64_t mask,
+        std::shared_ptr<ThreadConfigSetter> setter =
```

Comment:
The constructors should actually take in a unique_ptr of the ThreadConfigSetter and graduate it to a shared_ptr.